opm/simulators/flow/CompositionalContainer.cpp

```diff
@@ -0,0 +1,183 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+  Consult the COPYING file in the top-level source directory of this
+  module for the precise wording of the license and the list of
+  copyright holders.
+*/
+
+#include <config.h>
+#include <opm/simulators/flow/CompositionalContainer.hpp>
+
+#include <opm/material/fluidsystems/GenericOilGasFluidSystem.hpp>
+
+#include <opm/output/data/Solution.hpp>
+
+#include <algorithm>
+#include <tuple>
+
+#include <fmt/format.h>
+
+namespace Opm {
+
+template<class FluidSystem>
+void CompositionalContainer<FluidSystem>::
+allocate(const unsigned bufferSize,
+         std::map<std::string, int>& rstKeywords)
+{
+    if (rstKeywords["ZMF"] > 0) {
+        this->allocated_ = true;
+        rstKeywords["ZMF"] = 0;
```

Comment:
If you're changing this anyway, we could write it as
```C++
if (auto& zmf = rstKeywords["ZMF"]; zmf > 0) {
    this->allocated = true;
    zmf = 0;
    // ...
}
```
to avoid attempting to insert `"ZMF"` into the map twice.

Reply:
sure thing.